src/core/thread/mle_router.cpp

```diff
@@ -3334,7 +3334,7 @@ ThreadError MleRouter::RestoreChildren(void)
     for (uint8_t i = 0; ; i++)
     {
         Child *child;
-        otChildInfo childInfo;
+        ChildInfo childInfo;
         uint16_t length;
 
         length = sizeof(childInfo);
```

Comment:
Line 3343 should probably be the following to support backward compatibility (rolling back version):

```
VerifyOrExit(length >= sizeof(childInfo), error = kThreadError_Failed);
```